File: backend/src/main/java/com/church/attendance/entity/Attendance.java
package com.church.attendance.entity;

import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Table(name = "attendance", 
       uniqueConstraints = @UniqueConstraint(columnNames = {"user_id", "service_id"}))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Attendance {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "service_id", nullable = false)
    private Service service;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private AttendanceStatus status;
    
    @Column(nullable = false)
    private Double latitude;  // 출석 체크 시 사용자 위도
    
    @Column(nullable = false)
    private Double longitude;  // 출석 체크 시 사용자 경도
    
    @Column(nullable = false)
    private Double distance;  // 교회와의 거리(미터)
    
    @CreatedDate
    @Column(nullable = false, updatable = false)
    private LocalDateTime checkedAt;  // 출석 체크 시간
    
    @Column(length = 500)
    private String notes;  // 비고
    
    public enum AttendanceStatus {
        PRESENT,  // 정상 출석
        LATE,     // 지각
        ABSENT    // 결석
    }
}

File: flutter_app/lib/models/attendance.dart
class Attendance {
  final int id;
  final int userId;
  final String userName;
  final int serviceId;
  final String serviceName;
  final String status;
  final double latitude;
  final double longitude;
  final double distance;
  final DateTime checkedAt;
  final String? notes;

  Attendance({
    required this.id,
    required this.userId,
    required this.userName,
    required this.serviceId,
    required this.serviceName,
    required this.status,
    required this.latitude,
    required this.longitude,
    required this.distance,
    required this.checkedAt,
    this.notes,
  });

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      id: json['id'],
      userId: json['userId'],
      userName: json['userName'],
      serviceId: json['serviceId'],
      serviceName: json['serviceName'],
      status: json['status'],
      latitude: json['latitude'].toDouble(),
      longitude: json['longitude'].toDouble(),
      distance: json['distance'].toDouble(),
      checkedAt: DateTime.parse(json['checkedAt']),
      notes: json['notes'],
    );
  }

  String getStatusDisplayName() {
    switch (status) {
      case 'PRESENT':
        return '출석';
      case 'LATE':
        return '지각';
      case 'ABSENT':
        return '결석';
      default:
        return status;
    }
  }
}

